### PR TITLE
make the models in database fixed #24

### LINKED ISSUE #24: Database
# Database

## Database schema:
m e schema 2

## Database queries:

- [ ] get user by username and password:
Need to check if the user exists in database `users` table, to check if he is authenticated (on login route).

- [ ] Get activities:
Need to return all activities from `activities` table on the database.

- [ ] Add new activity
Need to create new activity and set it into `activities` table of the database.

- [ ] Add student to activity.
Needs to create a new row on the `activities_users` table, to relate the user to the training.

- [ ] Create new program
Needs to create (insert) a new program into `programs` table of the database.

- [ ] Add new user
Needs to create new user into `users` table of the database.

- [ ] Get programs 
Need to get list of programs from 'programs` table of the database.

- [ ] Get activity by id
Need to get activity details using its id.

- [ ] Get program by id
Need to view program details using its id

- [ ] Get activities by program id
Need to get list of activities that related to the program

- [ ] Get activities by objective id
Need to get list of activities that related to the objective.

- [ ] Get students by activity id.
Need to get all students related to activity id.

### PR DESCRIPTION
**Pull request by amanshawar**

### CODE REVIEW

File: client/.gitignore [hunk 1]
```diff
@@ -22,3 +22,4 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 node
+.env
```
Comment (m4v15) [inline client/.gitignore]: this should be config.env!

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+// const bodyParser = require('body-parser');
```
Comment (m4v15) [inline src/app.js]: get rid of these comments

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const path = require('path');
+// const router = require('./router.js');
+
+// app.use(bodyParser.urlencoded({ extended: false }));
+// app.use(bodyParser.json());
+// app.use(cookieParser());
+app.use(express.static(path.join(__dirname, '..', 'public')));
+// app.use(router);
+
+app.set('views', path.join(__dirname, 'views'));
```
Comment (m4v15) [inline src/app.js]: pretty sure you don't want this at all?

File: src/server.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const app = require('./app.js');
+require('env2')('config.env');
+
+app.set('PORT', process.env.PORT);
+app.listen(app.get('PORT'), () => {
+  // eslint-disable-next-line no-console
+  console.log(`The Server Is Up And Running On http://localhost:${app.get('PORT')}`);
+  // eslint-disable-next-line global-require
+  require('./mongoose.config');
```
Comment (m4v15) [inline src/server.js]: at the moment you are running your server, and then connecting to mongoose/mongo within the server. I would do this the other way around/restructure this `mongo.config` file:

```js
// mongo.config.js
const mongoose = require('mongoose');
require('env2')('config.env');

const mongoConnection = mongoose.connect(process.env.DATABASE_URL, { useNewUrlParser: true, useCreateIndex: true });

mongoConnection.connection.on('error', (err) => {
  // eslint-disable-next-line no-console
  console.log('mongoose connection error: ', err);
});
mongoConnection.connection.on('disconnected', () => {
  // eslint-disable-next-line no-console
  console.log('mongoose has been disconnected!');
});

module.exports = mongoConnection
```

```js
// server.js
const app = require('./app.js');
const db = require('./mongoose.config.js')
require('env2')('config.env');

db.connection.on('connected', () => {
  app.listen(app.get('PORT'), () => {
  // eslint-disable-next-line no-console
  console.log(`The Server Is Up And Running On http://localhost:${app.get('PORT')}`);
  })
});
```

Does this make sense?

See here as well: https://stackoverflow.com/questions/24474386/pass-mongoose-connection-to-module

Comment (m4v15) [inline src/server.js]: When requiring other javascript files, I would always add the `.js` extension, as you do for the `app` above. You can drop it, but if so, you should be dropping it everywhere (consistency is key)

File: package.json [hunk 5]
```diff
@@ -18,5 +18,17 @@
   "bugs": {
     "url": "https://github.com/FACK1/M-A-Platform/issues"
   },
-  "homepage": "https://github.com/FACK1/M-A-Platform#readme"
+  "homepage": "https://github.com/FACK1/M-A-Platform#readme",
+  "dependencies": {
+    "env2": "^2.2.2",
+    "eslint-config-airbnb": "^17.1.0",
```
Comment (m4v15) [inline package.json]: should this be in devDependencies?

File: src/router.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const express = require('express');
+
+const router = express.Router();
+
+router.get('/', (req, res) => {
+  res.send('hello world');
```
Comment (jema28) [inline src/router.js]: This response isn't necessary as the home route is handled in the frontend. You can change this to the * route that sends the react index.html back as @m4v15 mentioned last week. Here's an example: 
```node
app.get('*', (req, res) => {
    res.sendFile(path.join(__dirname, '..', 'client', 'build', 'index.html'));
  })
```

File: client/.gitignore [hunk 7]
```diff
@@ -22,4 +22,3 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 node
-.env
```
Comment (m4v15) [inline client/.gitignore]: ahhh sorry I thought this was you're server `.gitignore` - it looks like the client _does_ have a `.env` file so I guess you do need this line 🙈

**Pull request merged**